src/main/java/duke/DatedTask.java

```diff
@@ -0,0 +1,43 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Handles addition of date-based Tasks.
+ */
+
+public class DatedTask extends Task {
```

Comment:
Looking at this class, correct me if I'm wrong but if you would like Deadline and Event to inherit from DatedTask without DatedTask being instantiated, you could event make DatedTask abstract and inherit from Task.

Reply:
Indeed! Thanks for highlighting this.